src/prefect/orion/database/migrations/versions/postgresql/2023_01_26_045501_2882cd2df465_migrate_artifact_data.py

```diff
@@ -0,0 +1,125 @@
+"""Migrates state data to the artifact table
+
+Revision ID: 2882cd2df465
+Revises: 2882cd2df464
+Create Date: 2023-01-26 04:55:01.358638
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "2882cd2df465"
+down_revision = "2882cd2df464"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    ### START DATA MIGRATION
+
+    # insert nontrivial task run state results into the artifact table
+    def update_task_run_artifact_data_in_batches(batch_size, offset):
+        return f"""
+            INSERT INTO artifact (task_run_state_id, task_run_id, data)
+            SELECT id, task_run_id, data
+            FROM task_run_state
+            WHERE has_data IS TRUE
+            LIMIT {batch_size} OFFSET {offset};
```

Comment:
Do we need a stable `ORDER BY` clause here too?

Same question for `update_flow_run_artifact_data_in_batches`

Reply:
yes good catch